lib/miq_automation_engine/service_models/miq_ae_service_service.rb

```diff
@@ -48,6 +48,32 @@ def set_dialog_option(key, value)
       end
     end
 
+    def service_vars_options
+      @object.options[:service_vars] || HashWithIndifferentAccess.new
+    end
+
+    def get_service_vars_option(key)
+      service_vars_options[key]
+    end
+
+    def set_service_vars_option(key, value)
+      ar_method do
+        @object.options[:service_vars] ||= HashWithIndifferentAccess.new
```

Comment:
Can the `service_vars_options` method be reused here rather than duplicating it?

Reply:
This is the ```set``` method. The other one is the ```get``` method to retrieve data from DB.

Reply:
I think @bdunne is correct that this should reuse the `service_vars_options` method and that the `service_vars_options` method should use `||=` as mentioned above.  That would reduce the places that have to know the options hash key and the `HashWithIndifferentAccess` logic.